pkg/cloud/services/ec2/instances.go

```diff
@@ -182,20 +182,7 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte, use
 	}
 	input.SubnetID = subnetID
 
-	if ptr.Deref(scope.AWSMachine.Spec.PublicIP, false) {
-		subnets, err := s.getFilteredSubnets(&ec2.Filter{
-			Name:   aws.String("subnet-id"),
-			Values: aws.StringSlice([]string{subnetID}),
-		})
-		if err != nil {
-			return nil, fmt.Errorf("could not query if subnet has MapPublicIpOnLaunch set: %w", err)
-		}
-		if len(subnets) == 0 {
-			return nil, fmt.Errorf("expected to find subnet %q", subnetID)
-		}
-		// If the subnet does not assign public IPs, set that option in the instance's network interface
-		input.PublicIPOnLaunch = ptr.To(!aws.BoolValue(subnets[0].MapPublicIpOnLaunch))
-	}
+	input.PublicIPOnLaunch = ptr.To(ptr.Deref(scope.AWSMachine.Spec.PublicIP, false))
```

Comment:
why can't you assign `scope.AWSMachine.Spec.PublicIP` as is?

Reply:
yeah. the intention was to also force a value, but considering it will be used only [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4905/files#diff-4d2ea189a70ea097ffeeb50f78cc676a224386d4fe17b1bec9c83a4731d9c379R539), it will be better to assign as is. Thanks